jpos/src/main/java/org/jpos/iso/ISODate.java

```diff
@@ -131,7 +134,7 @@ public static Date parseDateTime(String s, TimeZone timeZone) {
      * @return Date
      */
     public static Date parseISODate (String d) {
-        return parseISODate (d, System.currentTimeMillis());
+        return parseISODate (d, Instant.now(NanoClock.systemUTC()).toEpochMilli());
```

Comment:
We are talking about YEAR accuracy here, makes no sense to use nanoseconds. This will just slow down a method heavily used in most messages going through the system.